### DISCUSSION TITLE
Pagination doesn´t work after search value

### DISCUSSION THREAD

**Discussion by inkael** [livewire/livewire#4171] [7. Bugs (archived)] [answered]
Hi, I have some troubles with pagination. basically at the beginning works well. I have a list of videos and I show them into component and paginated also I have input text for filtering videos and works well but when I erased the value from input . the paginate doesn´t response anymore.. Any Ideas?

**Comment by godismyjudge95** [comment]
It's common for pagination needing to be reset after searching/filtering since that changes the total number of results and thus the number of pages:  https://laravel-livewire.com/docs/2.x/pagination#resetting-pagination

Let me know if the above doesn't help you.

**Reply by inkael** [reply]
Well at the end  I just adapt my code. before I had like that 
` if  ($this->search !== null ){ $videos->$videos::where($colum,"like","%".$this->search."%");}`
So I delete IF and just left `$videos::where($colum,"like","%".$this->search."%")` and that worked.

**Comment by joshhanley** [comment, marked as answer]
@inkael the issue will probably be that `$this->search` won't be null when your clear the input field, instead it gets set to an empty string `$this->search = ''`, so you should base your check around that or make sure you convert it to null first if it is an empty string.

Hope this helps!

**Reply by inkael** [reply]
You're totally right, that was my mistake

**Reply by Melese999** [reply]
> @inkael the issue will probably be that `$this->search` won't be null when your clear the input field, instead it gets set to an empty string `$this->search = ''`, so you should base your check around that or make sure you convert it to null first if it is an empty string.
> 
> Hope this helps!

how to make it null

**Reply by joshhanley** [reply]
@Melese999 you can use an updating hook, and pass `$value` by reference to update it. For example (notice the `&` in front of `$value):

```php
public function updatingSearch(&$value)
{
    if ($value == '') {
        $value = null;
    }
}
```

Hope this helps!
